dtfs-central-api/src/v1/validation/utilisation-report-service/selected-fee-record-validator.test.ts

```diff
@@ -1,14 +1,28 @@
-import { Currency, FeeRecordEntity, FeeRecordEntityMockBuilder, UtilisationReportEntityMockBuilder } from '@ukef/dtfs2-common';
-import { validateSelectedFeeRecordsAllHaveSamePaymentCurrency } from './selected-fee-record-validator';
+import {
+  CURRENCY,
+  Currency,
+  FEE_RECORD_STATUS,
+  FeeRecordEntity,
+  FeeRecordEntityMockBuilder,
+  FeeRecordStatus,
+  PaymentEntity,
+  PaymentEntityMockBuilder,
+  UtilisationReportEntityMockBuilder,
+} from '@ukef/dtfs2-common';
+import { FeeRecordRepo } from '../../../repositories/fee-record-repo';
+import {
+  validateSelectedFeeRecordsAllHaveSamePaymentCurrency,
+  validateThatAllSelectedFeeRecordsWithPaymentsFormACompletePaymentGroup,
+} from './selected-fee-record-validator';
 import { InvalidPayloadError } from '../../../errors';
 
+jest.mock('../../../repositories/fee-record-repo');
+
 describe('selected fee record validator', () => {
   describe('validateSelectedFeeRecordsAllHaveSamePaymentCurrency', () => {
     it('does not throw if only one fee record', () => {
       // Arrange
-      const selectedFeeRecords = [
-        FeeRecordEntityMockBuilder.forReport(UtilisationReportEntityMockBuilder.forStatus('RECONCILIATION_IN_PROGRESS').build()).build(),
-      ];
+      const selectedFeeRecords = [FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).build()];
 
       // Act + Assert
       expect(() => validateSelectedFeeRecordsAllHaveSamePaymentCurrency(selectedFeeRecords)).not.toThrow();
```

Comment:
We can rewrite all these non throwing test cases but its not a big deal if you prefer this way :) 

```
    const nonThrowingTestCases = [
      {
        description: 'only one fee record',
        selectedFeeRecords: [FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).build()],
      },
      {
        description: 'no fee records provided',
        selectedFeeRecords: [],
      },
      {
        description: 'all fee records have the same payment currency',
        selectedFeeRecords: [aFeeRecordWithPaymentCurrency('GBP'), aFeeRecordWithPaymentCurrency('GBP')],
      },
    ];

    it.each(nonThrowingTestCases)(`does not throw if $description`, ({ selectedFeeRecords }) => {
      expect(() => validateSelectedFeeRecordsAllHaveSamePaymentCurrency(selectedFeeRecords)).not.toThrow();
    });
```

Reply:
I agree with Ruby's comment, I think we should leave as is